src/events.jl

```diff
@@ -0,0 +1,136 @@
+@with_kw mutable struct EventList{T1<:AbstractVector,T2<:AbstractVector,T3<:AbstractMatrix}
+    time::T1
+    energy::T2=Float64[]
+    ncounts::Int=0
+    mjdref::Float64=0
+    dt::Float64=0
+    notes::String=""
+    gti::T3=reshape(Float64[],0,2)
+    PI::Vector{Int64}=Int[]
```

Comment:
`Int` isn't necessarily `Int64` (it is on 64-bit systems, but not on 32-bit ones).  You either use `Int` or `Int64` everywhere, you have to be consistent.  If that helps in the choice, literal integers are of type `Int` (so they're actual type depends on the system where the code is run)